core/domain/state_domain.py

```diff
@@ -1641,6 +1807,66 @@ def from_dict(cls, written_translations_dict):
 
         return cls(translations_mapping)
 
+    def to_proto(self):
+        """Returns a proto representation of the written translations object.
+
+        Returns:
+            WrittenTranslationsDto. The proto object.
+        """
+        translation_language_mapping_protos = (
+            self._to_translation_language_mapping_proto(
+                self.translations_mapping))
+
+        return languages_pb2.WrittenTranslationsDto(
+            translation_language_mapping=translation_language_mapping_protos
+        )
+
+    @classmethod
+    def _to_translation_language_mapping_proto(cls, translations_mapping):
+        """Creates a WrittenTranslationContentMapping proto object.
+
+        Args:
+            translations_mapping: dict. A dict mapping the content Ids
+                to the dicts which is the map of abbreviated code of the
+                languages to WrittenTranslation objects.
+
+        Returns:
+            list. The WrittenTranslationContentMapping protos list.
+        """
+        language_to_content_id_written_translation_map = (
+            collections.defaultdict(dict))
+        for (content_id, language_code_to_written_translation) in (
+            translations_mapping.items()):
+            for (language_code, written_translation) in (
+                language_code_to_written_translation.items()):
+                written_translation_proto = (
+                    written_translation.to_proto())
+                language_to_content_id_written_translation_map[language_code][content_id] = written_translation_proto # pylint: disable=line-too-long
+
+        language_code_to_enum_map = {
+            'en': languages_pb2.LanguageType.ENGLISH,
+            'hi-en': languages_pb2.LanguageType.HINGLISH,
+            'ar': languages_pb2.LanguageType.ARABIC,
+            'hi': languages_pb2.LanguageType.HINDI,
+            'pt-br': languages_pb2.LanguageType.BRAZILIAN_PORTUGUESE
+        }
+        written_translation_content_mapping_protos_list = []
+
+        for (language_code, written_translation_content_map) in (
+            language_to_content_id_written_translation_map.items()):
+            if language_code in language_code_to_enum_map:
+                proto = languages_pb2.WrittenTranslationContentMappingDto(
+                    language=language_code_to_enum_map[language_code],
+                    translation_content_mapping=written_translation_content_map)
+            else:
+                proto = languages_pb2.WrittenTranslationContentMappingDto(
+                    language=(
```

Comment:
> Do we actually need to store the unspecified mappings with the proto?

It is designed in such a way that if any language is not found or available, then it is unspecified.
```proto
enum LanguageType {
  option (structure_enum_version_type) = LANGUAGE_PROTOS_VERSION;

  // The default language. This generally corresponds to cases when a language isn't defined, is
  // unsupported, or it's not known whether the language is supported. Clients are expected to
  // default to no-op behaviors when they encounter this language type.
  LANGUAGE_CODE_UNSPECIFIED = 0;

  // Corresponds to general English (that is, not particularly regionally tied). This is the assumed
  // default language for all content strings.
  ENGLISH = 1;

  // Corresponds to general Arabic (that is, not particularly regionally tied).
  ARABIC = 2;

  // Corresponds to general Hindi (that is, not particularly regionally tied).
  HINDI = 3;

  // Corresponds to the macaronic language Hinglish which combines Hindi and English. This
  // corresponds to a generic 'Hinglish' which combines localisms from different areas, but is not
  // predominantly tied to a particular region.
  HINGLISH = 4;

  // Corresponds to a regional Portuguese that's primarily used in Brazil.
  BRAZILIAN_PORTUGUESE = 5;
}
```

Reply:
Could you please confirm with @BenHenning what the expected behaviour around all these cases is? I understand the comment in the proto file but my reading is that it is saying how to interpret unspecified data. But it is not clear to me whether we need to be storing all that unspecified data in the proto and shipping it over the wire in the first place (seems wasteful IMO), and this needs to be determined.

Reply:
I think we should strip languages and voiceovers that aren't supported by the app rather than mapping to unspecified since the app isn't going to use that, anyway.

Reply:
done